kafka/config.go

```diff
@@ -175,27 +174,24 @@ func WithLogger(ctx context.Context, log logrus.FieldLogger) ConfigOpt {
 
 		// Read from channel and print logs using the provided logger.
 		go func() {
-			defer close(logsChan)
+			// Do not close logsChan because confluent-kafka-go will send logs until we close the client.
+			// Otherwise it will panic trying to send messages to a closed channel.
 			for {
-				select {
-				case <-ctx.Done():
+				m, ok := <-logsChan
```

Comment:
You could range over the channel as well instead.

Reply:
Sure, good catch! fixed